### PR TITLE
Use more enum flags

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Converting a few stray enums

### CODE REVIEW

File: src/OpenLoco/src/Windows/Construction/OverheadTab.cpp [hunk 1]
```diff
@@ -121,7 +121,7 @@ namespace OpenLoco::Ui::Windows::Construction::Overhead
     // 0x0049ECD1
     static void onUpdate(Window& self)
     {
-        Common::onUpdate(&self, (1 << 5));
+        Common::onUpdate(&self, GhostFlags::flag4);
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Windows/Construction/OverheadTab.cpp:124]: Is this a mistake? Looks like it should've been flag5?

Comment (duncanspumpkin) [reply src/OpenLoco/src/Windows/Construction/OverheadTab.cpp:124]: It was a mistake in the old code.

File: src/OpenLoco/src/Windows/Construction/Construction.h [hunk 2]
```diff
@@ -13,9 +13,20 @@ using namespace OpenLoco::World::TileManager;
 
 namespace OpenLoco::Ui::Windows::Construction
 {
+    enum class GhostFlags : uint8_t
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Windows/Construction/Construction.h:16]: i feel i'm repeating myself too much with these. Would `GhostVisibleFlags::arrow` make more sense?

Comment (AaronVanGeffen) [reply src/OpenLoco/src/Windows/Construction/Construction.h:16]: I agree that sounds better. Though perhaps `GhostVisibilityFlags`?

File: src/OpenLoco/src/Windows/Construction/Construction.h [hunk 3]
```diff
@@ -13,20 +13,20 @@ using namespace OpenLoco::World::TileManager;
...
+        station = 1U << 3,
+        overhead = 1U << 4,
     };
-    OPENLOCO_ENABLE_ENUM_OPERATORS(GhostFlags);
+    OPENLOCO_ENABLE_ENUM_OPERATORS(GhostVisibilityFlags);
 
     static loco_global<uint16_t[44], 0x004F891C> _trackPieceToFlags; // Roll into TrackData?
     static loco_global<uint8_t, 0x00522095> _byte_522095;
-    static loco_global<GhostFlags, 0x00522096> _byte_522096;
+    static loco_global<GhostVisibilityFlags, 0x00522096> _byte_522096;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Windows/Construction/Construction.h:29]: We should probably name `_byte_522096` while we're at it. Perhaps just `_ghostVisibilityFlags`?

File: src/OpenLoco/src/Audio/Audio.cpp [hunk 4]
```diff
@@ -472,7 +472,7 @@ namespace OpenLoco::Audio
     // 0x0048A4BF
     void playSound(Vehicles::Vehicle2or6* v)
     {
-        if (v->var_4A & 1)
+        if ((v->var_4A & Vehicles::SoundFlags::flag0) != Vehicles::SoundFlags::none)
```
Comment (ZehMatt) [inline src/OpenLoco/src/Audio/Audio.cpp:475]: Could rename var_4A to soundFlags

Comment (duncanspumpkin) [reply src/OpenLoco/src/Audio/Audio.cpp]: done

**Review by ZehMatt — APPROVED** [review]

**Pull request merged**